### PR TITLE
Added bulk update operation

### PR DESCRIPTION
**Pull request by blackshadev**
Added and use bulk operation in engine.

### CODE REVIEW

**Comment by Jeroen-G** [comment]
@blackshadev it needs a CS fix

File: src/Application/IndexAdapterInterface.php [hunk 1]
```diff
@@ -13,4 +15,6 @@ public function delete(string $index, string $id);
     public function flush(string $index);
 
     public function search(SearchCommandInterface $command): Results;
+
+    public function bulk(BulkOperationInterface $command);
```
Comment (Jeroen-G) [inline src/Application/IndexAdapterInterface.php:19]: Could this also do bulk deletes?

Comment (blackshadev) [reply src/Application/IndexAdapterInterface.php:19]: Yes it could

**Comment by blackshadev** [comment]
> @blackshadev it needs a CS fix

I did do that. I think something is wrong elsewhere where you used a union type, while php 7.4 doesn't accept it.

**Comment by Jeroen-G** [comment]
OK!

**Pull request merged**